src/consensus/pbft/libbyz/Replica.cpp

```diff
@@ -411,25 +417,38 @@ void Replica::playback_pre_prepare(ccf::Store::Tx& tx)
 
     LOG_TRACE_FMT("Storing pre prepare at seqno {}", seqno);
 
-    ledger_writer->write_pre_prepare(tx);
+    last_te_version = ledger_writer->write_pre_prepare(tx);
+
+    last_executed++;
 
     if (global_commit_cb != nullptr && executable_pp->is_signed())
     {
+      LOG_TRACE_FMT(
+        "Global_commit: {} {}",
+        executable_pp->get_ctx(),
+        executable_pp->seqno());
+      LOG_TRACE_FMT("Checkpointing for seqno {}", last_executed);
+
+      state.checkpoint(last_executed);
+      last_gb_version = executable_pp->get_ctx();
+      last_gb_seqno = executable_pp->seqno();
+
       global_commit_cb(
         executable_pp->get_ctx(), executable_pp->view(), global_commit_info);
     }
 
-    last_executed++;
-
-    if (last_executed % checkpoint_interval == 0)
+    if (executable_pp->is_signed() && f() > 0)
     {
       mark_stable(last_executed, true);
     }
+
     rqueue.clear();
   }
   else
   {
-    PBFT_ASSERT(false, "Merkle roots don't match in playback pre-prepare");
+    LOG_INFO_FMT(
```

Comment:
LOG_FAIL?

Reply:
No we still want LOG_INFO here, it is not failing, it will just rollback